go/src/koding/klient/machine/machinetest/machinetest.go

```diff
@@ -145,6 +153,17 @@ func (n *NilBuilder) WaitForBuild(timeout time.Duration) error {
 }
 
 // BuildsCount returns how many times Build method was invoked.
-func (n *NilBuilder) BuildsCount() int {
+func (n *ClientBuilder) BuildsCount() int {
 	return int(atomic.LoadInt64(&n.buildsN))
 }
+
+// WaitForContextClose waits until context is done. It times out after specified
+// duration.
+func WaitForContextClose(ctx context.Context, timeout time.Duration) error {
```

Comment:
i wanted to separate businesses logic from its tests. If I used `context.WithTimeout`:

```go
ctx, _ := context.WithTimeout(foobar.Context(), time.Second)
<-ctx.Done()
if err := ctx.Err(); err == context.DeadlineExceeded {
   // Context timed out either because test timeout or parent context timed out.
}
```
I would need to know the underlying logic to write correct tests. Even if i knew it, I wouldn't be able to distinguish who closed the context (parent context timed out(not an error) or testing context timed out(error)).

Currently, I'm only `cancel` context so the error will be `context.Cancelled` and `context.WithTimeout` solution is going to work. However, If anyone adds timeouts to the logic, this may trigger false negative tests results.